homeassistant/components/shelly/binary_sensor.py

```diff
@@ -241,10 +262,21 @@ def is_on(self) -> bool | None:
 class BlockSleepingBinarySensor(ShellySleepingBlockAttributeEntity, BinarySensorEntity):
     """Represent a block sleeping binary sensor."""
 
+    entity_description: BlockBinarySensorDescription
+
     @property
     def is_on(self) -> bool:
         """Return true if sensor state is on."""
         if self.block is not None:
             return bool(self.attribute_value)
 
         return self.last_state == STATE_ON
+
+    async def async_added_to_hass(self) -> None:
```

Comment:
Moved out of `ShellySleepingBlockAttributeEntity` as `state_class` doesn't apply outside sensor platform